File: auth-service/src/main/java/iuh/fit/authservice/config/MediaProperties.java
package iuh.fit.authservice.config;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@Getter
@Setter
@Component
@ConfigurationProperties(prefix = "media")
public class MediaProperties {

    public static final String DEFAULT_AVATAR_KEY = "static/default-avatar.jpg";

    private String publicBaseUrl = "";

    public String normalizedPublicBaseUrl() {
        String base = publicBaseUrl == null ? "" : publicBaseUrl.trim();
        while (base.endsWith("/")) {
            base = base.substring(0, base.length() - 1);
        }
        return base;
    }

    public String publicUrlForKey(String key) {
        return normalizedPublicBaseUrl() + "/" + key;
    }

    public String defaultAvatarUrl() {
        String base = normalizedPublicBaseUrl();
        if (base.isBlank()) {
            return "";
        }
        return publicUrlForKey(DEFAULT_AVATAR_KEY);
    }
}

File: media-service/src/config/mediaProperties.ts
import { env } from './env.js';

export function normalizedPublicBaseUrl(): string {
  let base = env.mediaPublicBaseUrl;
  while (base.endsWith('/')) {
    base = base.slice(0, -1);
  }
  return base;
}

export function publicUrlForKey(key: string): string {
  return `${normalizedPublicBaseUrl()}/${key}`;
}

export function presignExpirationSeconds(): number {
  return env.mediaPresignExpirationMinutes * 60;
}
